<?xml version="1.0"?>
<launch>

  <arg name="tcp_control_mode"	default="true"/>
  <arg name="ip"		default="10.66.171.52"/>
  <arg name="port"		default=""/>
  <arg name="device_name"	default="/tmp/ttyUR_a"/>
  <arg name="namespace"		default="gripper"/>
  <arg name="gripper_prefix"	default=""/>
  <arg name="config"		default="cmodel_action_controller"/>

  <group ns="$(arg namespace)">
    <!-- CModel TCP Controller -->
    <group if="$(arg tcp_control_mode)">
      <node name="cmodel_tcp_driver"
	    pkg="aist_robotiq" type="cmodel_tcp_driver.py"
	    args="$(arg ip) $(arg port)" output="screen" respawn="true"/>
    </group>
    <group unless="$(arg tcp_control_mode)">
      <node name="cmodel_rtu_driver"
	    pkg="aist_robotiq" type="cmodel_rtu_driver.py"
	    args="$(arg device_name)" output="screen" respawn="true"/>
    </group>

    <param name="gripper_prefix" value="$(arg gripper_prefix)"/>
    <rosparam file="$(dirname)/../config/$(arg config).yaml"/>
    <node name="cmodel_action_controller"
	  pkg="aist_robotiq" type="cmodel_action_controller.py"
	  output="screen" />
  </group>

</launch>
